backends/dpdk/dpdkArch.cpp

```diff
@@ -1527,6 +1541,87 @@ const IR::Node* DismantleMuxExpressions::postorder(IR::AssignmentStatement* stat
        }
   }
 */
+
+bool CopyMatchKeysToSingleStruct::isLearnerTable(const IR::P4Table* t) {
+    bool use_add_on_miss = false;
+    auto add_on_miss = t->properties->getProperty("add_on_miss");
+    if (add_on_miss == nullptr) return false;
+    if (add_on_miss->value->is<IR::ExpressionValue>()) {
+        auto expr = add_on_miss->value->to<IR::ExpressionValue>()->expression;
+        if (!expr->is<IR::BoolLiteral>()) {
+            ::error(ErrorType::ERR_UNEXPECTED,
+                    "%1%: expected boolean for 'add_on_miss' property", add_on_miss);
+            return false;
+        } else {
+            use_add_on_miss = expr->to<IR::BoolLiteral>()->value;
+        }
+    }
+    return use_add_on_miss;
+}
+
+int CopyMatchKeysToSingleStruct::getFieldSizeBits(const IR::Type *field_type) {
+    if (auto t = field_type->to<IR::Type_Bits>()) {
```

Comment:
have enums been eliminated?

Reply:
Yes. Enum elimination is done in midend.